src/Security/Infrastructure/Repository/AuthenticationRepository.php

```diff
@@ -0,0 +1,432 @@
+              provider_token.token AS provider_token,
+              provider_token.creation_date as provider_token_creation_date,
+              provider_token.expiration_date as provider_token_expiration_date,
+              refresh_token.token AS refresh_token,
+              refresh_token.creation_date as refresh_token_creation_date,
+              refresh_token.expiration_date as refresh_token_expiration_date
+            FROM `:db`.security_authentication_tokens sat
+            INNER JOIN `:db`.security_token provider_token ON provider_token.id = sat.provider_token_id
+            LEFT JOIN `:db`.security_token refresh_token ON refresh_token.id = sat.provider_token_refresh_id
+            WHERE sat.token = :token
+        "));
+        $statement->bindValue(':token', $token, \PDO::PARAM_STR);
+        $statement->execute();
+
+        if ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
+            $expirationDate = $result['provider_token_expiration_date'] !== null
+                ? (new \DateTime())->setTimestamp((int) $result['provider_token_expiration_date'])
+                : null;
+            $providerToken = new ProviderToken(
+                null,
+                $result['provider_token'],
+                (new \Datetime())->setTimestamp((int) $result['provider_token_creation_date']),
+                $expirationDate
+            );
+
+            $providerRefreshToken = null;
+            if ($result['refresh_token'] !== null) {
+                $expirationDate = $result['refresh_token_expiration_date'] !== null
+                    ? (new \DateTime())->setTimestamp((int) $result['refresh_token_expiration_date'])
+                    : null;
+                $providerRefreshToken = new ProviderToken(
+                    null,
+                    $result['refresh_token'],
+                    (new \Datetime())->setTimestamp((int) $result['refresh_token_creation_date']),
+                    $expirationDate
+                );
+            }
+
+            return new AuthenticationTokens(
+                (int) $result['user_id'],
+                (int) $result['provider_configuration_id'],
+                $token,
+                $providerToken,
+                $providerRefreshToken
+            );
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @throws \Assert\AssertionFailedException
+     */
+    public function findProvidersConfigurations(): array
+    {
+        $statement = $this->db->prepare($this->translateDbName("SELECT * FROM `:db`.provider_configuration"));
+        $statement->execute();
+        $providersConfigurations = [];
+        while ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
+            $providerConfiguration = (new ProviderConfiguration())
+                ->setId((int) $result['id'])
+                ->setType($result['type'])
+                ->setName($result['name'])
+                ->setForced((bool) $result['isForced'])
```

Comment:
Database columns name in snake case.
```suggestion
                ->setForced((bool) $result['is_forced'])
```